File: comms/push-notifications/list-push-notifications/list-push-notifications.php
<?php

namespace Example;

use Twilio\Comms\TwilioComms;
use Twilio\Comms\PushNotifications\Requests\PushNotificationsListRequest;
use Twilio\Comms\Types\PushNotificationStatus;
use DateTime;
use Twilio\Comms\Types\PushNotificationProvider;

$client = new TwilioComms(
    accountId: 'TWILIO_ACCOUNT_SID',
    authToken: 'TWILIO_AUTH_TOKEN',
);
$client->pushNotifications->list(
    new PushNotificationsListRequest([
        'operationId' => 'comms_operation_01h9krwprkeee8fzqspvwy6nq8',
        'status' => PushNotificationStatus::Scheduled->value,
        'tags' => 'key_1:value;key_2:value;',
        'startDate' => new DateTime('2024-01-15T09:30:00Z'),
        'endDate' => new DateTime('2024-01-15T09:30:00Z'),
        'user' => 'comms_pushnotificationuser_01h9krwprkeee8fzqspvwy6nq8',
        'provider' => PushNotificationProvider::Apn->value,
        'pageToken' => 'pageToken',
        'pageSize' => 50,
    ]),
);

File: comms/push-notifications/list-push-notifications/list-push-notifications.py
from twilio_comms import TwilioComms
import datetime

client = TwilioComms(
    account_id="TWILIO_ACCOUNT_SID",
    auth_token="TWILIO_AUTH_TOKEN",
)

client.push_notifications.list(
    operation_id="comms_operation_01h9krwprkeee8fzqspvwy6nq8",
    status="SCHEDULED",
    tags="key_1:value;key_2:value;",
    start_date=datetime.datetime.fromisoformat("2024-01-15T09:30:00+00:00"),
    end_date=datetime.datetime.fromisoformat("2024-01-15T09:30:00+00:00"),
    user="comms_pushnotificationuser_01h9krwprkeee8fzqspvwy6nq8",
    provider="APN",
    page_token="pageToken",
    page_size=50,
)
